test/tests.py

```diff
@@ -0,0 +1,89 @@
+from representatives_votes.contrib.parltrack import import_dossiers
+from representatives_votes.contrib.parltrack import import_single_dossier
+from representatives_votes.contrib.parltrack import import_votes
+from representatives_votes.models import Dossier, Proposal, Vote
+
+from .parltrack_stub import TestDataSource
+
+
+class ImportTest(TestCase):
+    def setUp(self):
+        for model in (Representative, Dossier, Proposal, Vote):
+            model.objects.all().delete()
+
+    def _get_datapath(self, file):
+        return os.path.join(os.path.dirname(__file__), 'data', file)
+
+    def _test_import(self, scenario, callback):
+        fixture = self._get_datapath('%s_fixture.json' % scenario)
+        expected = self._get_datapath('%s_expected.json' % scenario)
+
+        # Disable django auto fields
+        exclude = ('id', '_state', 'created', 'updated')
+
+        with open(fixture, 'r') as f:
+            callback(f)
+
+        with open(expected, 'r') as f:
+            for obj in Deserializer(f.read()):
+                compare = copy.copy(obj.object.__dict__)
+
+                for f in exclude:
+                    if f in compare:
+                        compare.pop(f)
+
+                type(obj.object).objects.get(**compare)
+
+    def test_parltrack_import_dossiers(self):
+        with self.assertNumQueries(15):
+            self._test_import('dossiers', import_dossiers.main)
+
+    def test_parltrack_import_single_dossier(self):
+        call_command('loaddata', os.path.join(os.path.abspath(
+            parltrack.__path__[0]), 'tests', 'representatives_expected.json'))
+
+        with self.assertNumQueries(16):
+            self._test_import('single', import_single_dossier.main)
+
+    def test_parltrack_import_votes(self):
+        call_command('loaddata', os.path.join(os.path.abspath(
+            parltrack.__path__[0]), 'tests', 'representatives_expected.json'))
+        call_command('loaddata', self._get_datapath('dossiers_expected.json'))
+
+        with self.assertNumQueries(65):
+            self._test_import('votes', import_votes.main)
+
+    def test_dossier_sync(self):
+        call_command('loaddata', self._get_datapath('sync_initial.json'))
+
+        dossier = Dossier.objects.get(reference='2012/2002(INI)')
+
+        with self.assertNumQueries(3):
+            dossier.sync(True, TestDataSource())
+
+        exp = "Agenda for change: the future of EU development policy"
+        assert dossier.title == exp
+
+    def test_dossier_sync_norefresh(self):
+        call_command('loaddata', self._get_datapath('sync_initial.json'))
+
+        dossier = Dossier.objects.get(reference='2012/2002(INI)')
+
+        with self.assertNumQueries(2):
+            dossier.sync(False, TestDataSource())
+
+        assert dossier.title == 'initial title'
+
+        dossier.refresh_from_db()
+
+        exp = "Agenda for change: the future of EU development policy"
+        assert dossier.title == exp
```

Comment:
unittest

![2016-01-03-212433_621x385_scrot](https://cloud.githubusercontent.com/assets/94636/12080626/7c23971a-b260-11e5-9027-8c4c38d7e1d3.png)

pytest:

![2016-01-03-212442_873x538_scrot](https://cloud.githubusercontent.com/assets/94636/12080629/8d3c3f8e-b260-11e5-861d-d5436f75af49.png)

What's best ? no color + self.assert* everywhere or pytest ?

Note that you can also use unittest-style classes with pytest if that's what you're after !